lxd/cluster/membership.go

```diff
@@ -443,64 +450,72 @@ func Join(state *state.State, gateway *Gateway, networkCert *shared.CertInfo, se
 	err = state.Cluster.ExitExclusive(func(tx *db.ClusterTx) error {
 		node, err := tx.GetPendingNodeByAddress(address)
 		if err != nil {
-			return errors.Wrap(err, "failed to get ID of joining node")
+			return errors.Wrap(err, "Failed to get ID of joining node")
 		}
+
 		state.Cluster.NodeID(node.ID)
 		tx.NodeID(node.ID)
 
 		// Storage pools.
 		ids, err := tx.GetNonPendingStoragePoolsNamesToIDs()
 		if err != nil {
-			return errors.Wrap(err, "failed to get cluster storage pool IDs")
+			return errors.Wrap(err, "Failed to get cluster storage pool IDs")
 		}
+
 		for name, id := range ids {
 			err := tx.UpdateStoragePoolAfterNodeJoin(id, node.ID)
 			if err != nil {
-				return errors.Wrap(err, "failed to add joining node's to the pool")
+				return errors.Wrap(err, "Failed to add joining node's to the pool")
 			}
 
 			driver, err := tx.GetStoragePoolDriver(id)
 			if err != nil {
-				return errors.Wrap(err, "failed to get storage pool driver")
+				return errors.Wrap(err, "Failed to get storage pool driver")
 			}
 
 			if shared.StringInSlice(driver, []string{"ceph", "cephfs"}) {
 				// For ceph pools we have to create volume
 				// entries for the joining node.
 				err := tx.UpdateCephStoragePoolAfterNodeJoin(id, node.ID)
 				if err != nil {
-					return errors.Wrap(err, "failed to create ceph volumes for joining node")
+					return errors.Wrap(err, "Failed to create ceph volumes for joining node")
 				}
 			} else {
 				// For other pools we add the config provided by the joining node.
 				config, ok := pools[name]
 				if !ok {
-					return fmt.Errorf("joining node has no config for pool %s", name)
+					return fmt.Errorf("Joining member has no config for pool %s", name)
 				}
+
 				err = tx.CreateStoragePoolConfig(id, node.ID, config)
 				if err != nil {
-					return errors.Wrap(err, "failed to add joining node's pool config")
+					return errors.Wrap(err, "Failed to add joining node's pool config")
 				}
 			}
 		}
 
 		// Networks.
-		ids, err = tx.GetNonPendingNetworkIDs()
+		netids, err := tx.GetNonPendingNetworkIDs()
 		if err != nil {
-			return errors.Wrap(err, "failed to get cluster network IDs")
+			return errors.Wrap(err, "Failed to get cluster network IDs")
 		}
-		for name, id := range ids {
-			config, ok := networks[name]
-			if !ok {
-				return fmt.Errorf("joining node has no config for network %s", name)
-			}
-			err := tx.NetworkNodeJoin(id, node.ID)
-			if err != nil {
-				return errors.Wrap(err, "failed to add joining node's to the network")
-			}
-			err = tx.CreateNetworkConfig(id, node.ID, config)
-			if err != nil {
-				return errors.Wrap(err, "failed to add joining node's network config")
+
+		for _, network := range netids {
+			for name, id := range network {
+				config, ok := networks[name]
+				if !ok {
+					return fmt.Errorf("Joining member has no config for network %s", name)
```

Comment:
@C-Stout ideally this error message should be updated to use `%q` rather than `%s` and also include the project name using `%q` too. This way if a network join does fail we know which project the network is in.